internal/services/fileService.go

```diff
@@ -149,44 +167,89 @@ func (fs *fileService) StoreSolutionResult(solutionResult solution.SolutionResul
 		return err
 	}
 
-	file, err := os.Open(archiveFilePath)
+	return fs.sendArchiveToStorage(archiveFilePath, userID, taskID, submissionNumber)
+}
+
+func (fs *fileService) moveCompilationError(taskDir, outputDir string) error {
+	src := filepath.Join(taskDir, constants.CompileErrorFileName)
+	dst := filepath.Join(outputDir, constants.CompileErrorFileName)
+	return os.Rename(src, dst)
+}
+
+func (fs *fileService) sendArchiveToStorage(
+	archiveFilePath string,
+	userID, taskID, submissionNumber int64,
+) error {
+	body, contentType, err := fs.prepareMultipartBody(archiveFilePath, userID, taskID, submissionNumber)
+	if err != nil {
+		return err
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	req, err := fs.createRequestWithContext(ctx, body, contentType)
 	if err != nil {
 		return err
 	}
+
+	return fs.sendRequestAndHandleResponse(req)
+}
+
+func (fs *fileService) prepareMultipartBody(
+	archiveFilePath string,
+	userID, taskID, submissionNumber int64,
+) (*bytes.Buffer, string, error) {
```

Comment:
Do you need to return contentType if you are preparing multipartbody? By definition you now what type of body it is